test/src/test/java/jenkins/model/NodesTest.java

```diff
@@ -239,4 +244,44 @@ protected boolean allowLoad(@NonNull Node node) {
         }
     }
 
+    @Test
+    public void listenersCalledOnSetNodes() throws URISyntaxException, IOException, Descriptor.FormException {
+        var agentA = new DumbSlave("nodeA", "temp", r.createComputerLauncher(null));
+        var agentB = new DumbSlave("nodeB", "temp", r.createComputerLauncher(null));
+        var agentA2 = new DumbSlave("nodeA", "temp2", r.createComputerLauncher(null));
+        Jenkins.get().setNodes(List.of(agentA, agentB));
+        assertThat(CheckSetNodes.created, containsInAnyOrder(agentA.toString(), agentB.toString()));
```

Comment:
Test would be clearer using string constants rather than `Node.toString` I think, e.g.
```suggestion
        assertThat(CheckSetNodes.created, containsInAnyOrder("nodeA", "nodeB"));
```